File: Binary二进制/leetcode_1404_numSteps.py
class Solution:
    def numSteps(self, s: str) -> int:
        s = list(s)
        res = 0
        while len(s) and s!=['1']:
            if s[-1] == '0':
                s.pop()
            else:
                indx = len(s)-1
                while indx >= 0 and s[indx] == '1':
                    s[indx] = '0'
                    indx -= 1
                if indx >= 0:
                    s[indx] = '1'
                else:
                    s = ['1']+s
            res += 1
        return res

if __name__ == "__main__":
    sl = Solution()
    s = "1101"
    print(sl.numSteps(s))

File: 02.Binary二进制/leetcode_1404_numSteps.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int numSteps(string s) {
        int res = 0;
        while (s.size() > 1){
            if (s[s.size()-1] == '0'){
                s = s.substr(0, s.size()-1);
            }
            else{
                int indx = s.size()-1;
                while (indx >= 0 && s[indx] == '1'){
                    s[indx] = '0';
                    indx--;
                }
                if (indx >= 0){
                    s[indx] = '1';
                }
                else{
                    s = "1"+s;
                }
            }
            res++;
        }
        return res;
    }
};

int main(){
    Solution sl;
    string s = "1101";
    cout << sl.numSteps(s) << endl;
}
